pkg/xds/envoy/clusters/clusters.go

```diff
@@ -48,8 +48,20 @@ func CreateEdsCluster(ctx xds_context.Context, clusterName string, metadata *cor
 				},
 			},
 		},
-		TlsContext: tlsContext,
-	}), nil
+	})
+	if tlsContext != nil {
+		pbst, err := ptypes.MarshalAny(tlsContext)
+		if err != nil {
+			return nil, err
+		}
+		cluster.TransportSocket = &envoy_core.TransportSocket{
+			Name: "envoy.transport_sockets.tls",
```

Comment:
I guess we also should update `go-control-plane` library and use `envoy_wellknown.TransportSocketTls` constant for it.

Reply:
Good point, we should update to the newest version of go-control-plane, but I'd rather have this in the separate PR. We might get into some issues and this PR is big already.
Let me create a card for this.